### PR TITLE
InserOrReplace and DeleteIfExist with preconditions check have been added

### PR DESCRIPTION
**Pull request by KonstantinRyazantsev**

### CODE REVIEW

File: src/Lykke.AzureStorage/Tables/AzureTableStorage.cs [hunk 1]
```diff
@@ -435,6 +435,46 @@ IEnumerator IEnumerable.GetEnumerator()
...
+                        await table.ExecuteAsync(TableOperation.Replace(entity), GetRequestOptions(), null);
+                    }
+                    else
+                    {
+                        await table.ExecuteAsync(TableOperation.Insert(entity), GetRequestOptions(), null);
+                    }
+
+                    return true;
+                }
+                catch (StorageException e) when (e.RequestInformation.HttpStatusCode == 412)
```
Comment (starkmsu) [inline src/Lykke.AzureStorage/Tables/AzureTableStorage.cs]: Multiple catches can be merged into one with useing || for catch condition:
e.RequestInformation.HttpStatusCode == 412 || e.RequestInformation.HttpStatusCode == 409 || e.RequestInformation.HttpStatusCode == 404

File: src/Lykke.AzureStorage/Tables/AzureTableStorage.cs [hunk 2]
```diff
@@ -435,6 +435,46 @@ IEnumerator IEnumerable.GetEnumerator()
...
+                {
+                }
+                catch (StorageException e) when (e.RequestInformation.HttpStatusCode == 409)
+                {
+                }
+                catch (StorageException e) when (e.RequestInformation.HttpStatusCode == 404)
```
Comment (starkmsu) [inline src/Lykke.AzureStorage/Tables/AzureTableStorage.cs]: 404 - from this - https://docs.microsoft.com/en-us/rest/api/storageservices/table-service-error-codes - 404 is returned only for 'Table is not found' case. May be we need to throw exception in this case, right?

Comment (NovichikhinAlexey) [reply src/Lykke.AzureStorage/Tables/AzureTableStorage.cs]: good comment, need check other methods for this

Comment (KonstantinRyazantsev) [reply src/Lykke.AzureStorage/Tables/AzureTableStorage.cs]: Yes, but here - https://docs.microsoft.com/en-us/rest/api/storageservices/common-rest-api-error-codes there is ResourceNotFound. Replace can responce 404, if row has been deleted by another thread.

File: src/Lykke.AzureStorage/Tables/AzureTableStorage.cs (hunk 1 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.AzureStorage/Tables/AzureTableStorage.cs]: It's a mater of taste, but ok, I have changed

**Pull request merged**